lib/MC/MCWin64EH.cpp

```diff
@@ -206,10 +207,16 @@ static void EmitUnwindInfo(MCStreamer &streamer, WinEH::FrameInfo *info) {
   if (flags & (Win64EH::UNW_ChainInfo << 3))
     EmitRuntimeFunction(streamer, info->ChainedParent);
   else if (flags &
-           ((Win64EH::UNW_TerminateHandler|Win64EH::UNW_ExceptionHandler) << 3))
-    streamer.EmitValue(MCSymbolRefExpr::create(info->ExceptionHandler,
-                                              MCSymbolRefExpr::VK_COFF_IMGREL32,
-                                              context), 4);
+           ((Win64EH::UNW_TerminateHandler|Win64EH::UNW_ExceptionHandler) << 3)) {
+    if (classifyEHPersonality(info->ExceptionHandler->getName()) == EHPersonality::CoreCLR) {
```

Comment:
You could instead check if the runtime environment is CoreCLR (and assert if you see any unexpected personality routine if so).